api/src/test/java/org/openmrs/module/insuranceclaims/api/dao/impl/InsuranceClaimDiagnosisDaoImplTest.java

```diff
@@ -0,0 +1,38 @@
+package org.openmrs.module.insuranceclaims.api.dao.impl;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.insuranceclaims.api.dao.InsuranceClaimDiagnosisDao;
+import org.openmrs.module.insuranceclaims.api.model.InsuranceClaimDiagnosis;
+import org.openmrs.module.insuranceclaims.api.mother.InsuranceClaimDiagnosisMother;
+import org.openmrs.test.BaseModuleContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static org.hamcrest.Matchers.hasProperty;
+import static org.hamcrest.Matchers.is;
+
+public class InsuranceClaimDiagnosisDaoImplTest extends BaseModuleContextSensitiveTest {
+
+	@Autowired
+	private InsuranceClaimDiagnosisDao dao;
+
+	@Test
+	public void saveInsuranceClaimDiagnosis_shouldSaveAllPropertiesInDb() {
+		InsuranceClaimDiagnosis diagnosis = InsuranceClaimDiagnosisMother.createTestInstance();
+
+		dao.saveInsuranceClaimDiagnosis(diagnosis);
+
+		Context.flushSession();
+		Context.clearSession();
+
+		InsuranceClaimDiagnosis savedDiagnosis = dao.getInsuranceClaimDiagnosisByUuid(diagnosis.getUuid());
+
+		Assert.assertThat(savedDiagnosis, hasProperty("uuid", is(diagnosis.getUuid())));
+		Assert.assertThat(savedDiagnosis, hasProperty("concept", is(diagnosis.getConcept())));
+		Assert.assertEquals(diagnosis.getInsuranceClaim(), savedDiagnosis.getInsuranceClaim());
```

Comment:
we should use one convention assertEquals or assertThat

Reply:
changed